implementations/rust/ockam/ockam_transport_tcp/src/portal/portal_receiver.rs

```diff
@@ -39,12 +42,20 @@ impl Processor for TcpPortalRecvProcessor {
         };
 
         if len != 0 {
-            let mut vec = vec![0u8; len];
-            vec.copy_from_slice(&buf[..len]);
-            let msg = PortalInternalMessage::Payload(vec);
+            // Loop just in case buf was extended (should not happen though)
+            loop {
+                let at = min(MAX_PAYLOAD_SIZE, buf.len());
+                let chunk = buf.split_to(at);
+
+                let msg = PortalInternalMessage::Payload(chunk.to_vec());
```

Comment:
Could we change `PortalInternalMessage::Payload` (and `PortalMessage::Payload`) to accept `Bytes` instead of `Vec<u8>`? This would require the `serde` feature of `bytes` to be enabled but we would not have to copy the buffer contents here.

Reply:
Yeah, this is a good idea (use of bytes in more cases throughout would probably be good, although I suppose we'll need to be careful about this on embedded), but we should not do it as part of this change.

Reply:
Basically.